sci-mathematics/rstudio-1.1.383

waebbl opened the pull request:
Bug https://bugs.gentoo.org/633936 has updated to 1.1.368 and also mentions 1.1.383 but it's not included in the tree yet, although the bug has been closed.
I've used 1.1.153 as a basis and the comment from the above patch. The ebuild compiles and works for me.
Any feedback is welcome.

a17r commented:
Look at the diff between `1.0.153` and `1.1.368`; why do you use an older base than available for your version bump?

waebbl commented:
Thanks for the tip on rebasing, I'm still quite new to git and need to learn much. I'll look into how to properly work with rebasing.

A few explanations on why things are they way they are.

As for basing off of 1.0.153: When I created the commit, 1.1.368 wasn't out. I actually based my ebuild off of an earlier version of 1.1.368 from https://bugs.gentoo.org/622254. I offered the possibility of preparing a PR in https://bugs.gentoo.org/633936#c4 on Oct 15th and 1.1.368 was published on Oct 24th. When 1.1.368 came out, I had still not gotten any response to my offer. After discussing whether to just create a PR or not with a friend, I decided to give it a try and created the PR from the then ~10 days old commit.

As for the differences between 1.0.153 and 1.1.368: Most of them are in my ebuild too, as I put into account the comments in bugs #633936 and #622254 on gentoo's bugzilla. This illustrates why I had removed one of the patches (comment 633936#c2) and added another, which is not included in 1.1.368, but was in the ebuild in bug #622254 (see https://bugs.gentoo.org/attachment.cgi?id=494772). I also had not to pull in qtwebchannel and it compiled fine, but this might be because I already had it installed. I currently don't have a clean and sane testing environment to check out building in any detail.
The difference in src_install I must have missed, and the differences belonging to xdg-utils.eclass... well I must have been assuming exported functions do have a prefix of their eclass names. Strangely it seemed have to work, although I don't have the log file anymore to verify this.

I go and mix in those differences, so the ebuild is actually based on published the 1.1.368 ebuild.

waebbl commented:
Please note, the patch rstudio--0.98.1091-boost-1.57.patch, which I had removed in my ebuild, no longer applies. This is due to an added `#include <QProcess>` in `src/cpp/desktop/DesktopApplicationLaunch.hpp`, see [this](https://github.com/rstudio/rstudio/commit/585f7244797814d969418d8fe795d1952d84f6df#diff-707d3db8ed803af071e68fd80e87bec9) commit from Sep 26th. Also according to the bug report on https://bugreports.qt.io/browse/QTBUG-22829 which is referenced in the patch the related bug seems only applicable to Qt:4, but rstudio is actually using Qt:5.
So I'm gonna remove this patch in my ebuild. I also found the patch rstudio-1.1.357-boost.patch is not needed and removed it too.
